test_runner/performance/test_logical_replication.py

```diff
@@ -340,3 +340,104 @@ def test_publisher_restart(
+def test_snap_files(
+    pg_bin: PgBin,
+    neon_api: NeonAPI,
+    pg_version: PgVersion,
+    zenbenchmark: NeonBenchmarker,
+):
+    """
+    Creates a node with a replication slot. Generates pgbench into the replication slot,
+    then runs pgbench inserts while generating large numbers of snapfiles. Then restarts
+    the node and tries to peek the replication changes.
+    """
+    test_duration_min = 60
+    test_interval_min = 5
+    pgbench_duration = f"-T{test_duration_min * 60 * 2}"
+
+    project = neon_api.create_project(pg_version)
+    project_id = project["project"]["id"]
+    endpoint_id = project["endpoints"][0]["id"]
+    neon_api.wait_for_operation_to_finish(project_id)
+    error_occurred = False
+    try:
+        env = connection_parameters_to_env(project["connection_uris"][0]["connection_parameters"])
+        connstr = project["connection_uris"][0]["connection_uri"]
+        pg_bin.run_capture(["pgbench", "-i", "-s100"], env=env)
+
+        conn = psycopg2.connect(connstr)
+        conn.autocommit = True
+        with conn.cursor() as cur:
+            cur.execute(
+                "SELECT 'init' FROM pg_create_logical_replication_slot('slot', 'test_decoding')"
+            )
+        conn.close()
+
+        workload = pg_bin.run_nonblocking(
+            ["pgbench", "-c10", pgbench_duration, "-Mprepared"], env=env
+        )
+        try:
+            start = time.time()
+            while time.time() - start < test_duration_min * 60:
+                with psycopg2.connect(connstr) as conn:
+                    with conn.cursor() as cur:
+                        cur.execute(
+                            """
+                        SELECT data FROM
+                        pg_logical_slot_get_changes(
+                            'slot',
+                            NULL,
+                            NULL,
+                            'include-xids',
+                            '0',
+                            'skip-empty-xacts',
+                            '1')
+                        """
+                        )
+
+                check_pgbench_still_running(workload)
+                workload.terminate()
+
+                neon_api.restart_endpoint(
+                    project_id,
+                    endpoint_id,
+                )
+                neon_api.wait_for_operation_to_finish(project_id)
+
+                workload = pg_bin.run_nonblocking(
+                    ["pgbench", "-c10", pgbench_duration, "-Mprepared"],
+                    env=env,
+                )
+
+                time.sleep(test_interval_min * 60)
+
+                with psycopg2.connect(connstr) as conn:
+                    with conn.cursor() as cur:
+                        cur.execute(
+                            """
+                        select
+                        count(*)
+                        from
+                            (select pg_export_snapshot() from generate_series(1, 100000) g)
```

Comment:
Is it what generates a lot of snap files? I'm not familiar enough with LR internals, so I wonder why it's only the case when there is an active LR?

Reply:
Hmm , I believe these are different kind of snapshot files. Logical replication creates "snapshot files" under `pg_snapshots`, and pg_export_snapshot() creates files under `pg_logical/snapshots/`.

Reply:
This stackoverflow seems to say it stores it in `pg_snapshots`: https://stackoverflow.com/questions/73371963/what-the-pg-snapshots-folder-is-used-for

Reply:
`select count(*) from pg_ls_logicalsnapdir();` returns 209 right now, so I am creating SOME of these snapshot files now

Reply:
Ok we hvae it figured out